Aggiunta prima stesura per capitolo formazione (#124)
Il-Libro-Open-Source/book: docs/formazione/README.md

Basta una semplice ricerca su Google, del tipo: _"Come diventare un programmatore"_

Da lì in avanti i vari canali social, da Instagram a TikTok, dai video promozionali di YouTube alle inserzioni su Facebook, inizieranno a proporre differenti percorsi formativi, principalmente incentrati sulla figura professionale del _Fullstack Developer_.

Ma chi è questo fantomatico _Fullstack Developer_ e come mai è così ricercato dal mercato tanto da essere sulla bocca della maggior parte dei percorsi formativi?

Il _Fullstack Developer_ è uno sviluppatore che, teoricamente, conosce uno o più linguaggi di _frontend_ e di _backend.
Ok, qui si apre un sipario enorme che però cercherò di ridurre molto semplicemente come segue.
Stiamo parlando di **formazione** in ambito **programmazione**, quindi di quali strade si possono percorrere per imparare a programmare, e dunque diventare dei programmatori.
Quindi il programmatore crea il programma.
Il programma è composto da una parte grafica ed una parte logica.

Prendiamo l'esempio dell'ambito più semplice (in termini di comprensione): il web.

> Un programma sul web à un _sito web_, il quale è composto da una parte grafica, che permette all'utente di interagire, ed una parte logica, che a sua volta interpreta le azioni dell'utente.

Reviewer: à -> è

**Google è il nostro migliore amico**.

## Seconda fase: il tempo

Alcuni percorsi di formazione propongono 3 mesi altri 6 mesi, altri invece hanno dei pacchetti personalizzati.
Alcuni sono costituiti da classi e le lezioni si tengono in presenza, altri invece completamente da remoto con lezioni live oppure on-demand.
Anche qui non c'è una formula magica.

Diventare programmatore non è semplice, essere programmatore è una questione di **mindset**, quindi serve del tempo per cambiare il nostro modo di approcciare e di vedere le cose.
In sostanza... ci vuole tempo.

Quindi maggiore tempo dedichiamo alla nostra formazione e migliore è il risultato che otterremo alla fine del processo.
La scelta dovrebbe essere soggettiva e ponderata, basandosi sulla quantità di tempo che si ha a disposizione.
Il tempo è il maggiore investimento in questa fase, più tempo dedicate migliore sarà il risultato finale.

"Va bene, quindi quanto tempo serve e quanto tempo dovrei dedicare alla mia formazione?"
Il tempo non è mai abbastanza, quindi la risposta giusta è: il più possibile.

> Ad esempio: se il percorso di formazione offre lezioni mattutine, sarà necessario dedicare almeno altre 3 ore di pratica e approndimenti.

Reviewer: Typo: approfondimenti

# Formazione

Quando si muovono i primi passi, quelli dettati dalla curiosità e dalla voglia di intraprendere un nuovo percorso formativo/lavorativo, ci si imbatte sin da subito con il mondo della **formazione**.

Basta una semplice ricerca su Google, del tipo: _"Come diventare un programmatore"_

Da lì in avanti i vari canali social, da Instagram a TikTok, dai video promozionali di YouTube alle inserzioni su Facebook, inizieranno a proporre differenti percorsi formativi, principalmente incentrati sulla figura professionale del _Fullstack Developer_.

Ma chi è questo fantomatico _Fullstack Developer_ e come mai è così ricercato dal mercato tanto da essere sulla bocca della maggior parte dei percorsi formativi?

Il _Fullstack Developer_ è uno sviluppatore che, teoricamente, conosce uno o più linguaggi di _frontend_ e di \_backend.

> Ok, qui si apre un sipario enorme che però cercherò di ridurre molto semplicemente come segue.

Reviewer 2: Non sono sicuro della forma "io parlo a te", cioè sono d'accordo sul parlare al lettore, ma direi di provare a parlare _con_ lui, quindi proporrei un "...che cercheremmo di ridurre..."

Possiamo dire che è una spiegazione un pò approssimativa ma per ora dovrebbe bastare.

Quindi, per sommi capi, il _Fullstack Developer_ è la fusione tra il _Frontend Developer_ ed il _Backend Developer_.

I professionisti di settore sono ben consapevoli che ad oggi, con l'evoluzione dei linguaggi di programmazione, risulta **difficilissimo** conoscere e padroneggiare in maniera corretta ed approfondita entrambi gli aspetti (_frontend_ e _backend_), però i percorsi di formazione mirano piuttosto a fornire un'infarinatura generale di entrambi gli aspetti per permettere poi al futuro "programmatore" di scegliere in autonomia quale strada seguire.

"Ok, ma queste informazioni come possono essermi utili nella **scelta degli argomenti su cui formarmi?**"

Sapere che esistono questi 2 mondi, _frontend_ e _backend_, permette di fare una piccolissima cernita tra le varie strade che si possono intraprendere nel diventare **programmatore**.

La cosa fondamentale che bisogna tenere a mente quando si inizia ad intraprendere un percorso di formazione nell'ambito della programmazione è: se inizi con una cosa non è detto che poi farai quella per tutto il resto della carriera.

"Cosa vuol dire?"

> Ad esempio: se inizi come _Frontend Developer_ non è detto che un domani tu non possa cambiare strada e passare al lato osscuro (_backend_).

Reviewer 3: `oscuro` anziché `osscuro`

Per tale motivo spesso chi fa formazione propone dei percorsi di _fullstack_ in ambito web.

"Ok, ma come mai parliamo di siti web e non di videogiochi?"
Perchè la curva di apprendimento è più bassa nel caso dei siti web, quindi è più facile che sia a portata di molta più gente, e se teniamo a mente sempre il concetto fondamentale per cui "se si inizia come _frontend_ non è detto che si finisca da _frontend_" allora vien da sè pensare che un _frontend_ oggi possa diventare un _Game Developer_ domani.

Inoltre le tecnologie che si toccano con mano durante la fase di apprendimento in ambito web sono diverse (HTML, CSS, JavaScript) e ciò permette di avere una overview più ampia.

> Facciamo un esempio: con JavaScript possiamo sviluppare _siti web_ ma non solo, grazie alla versatilità del linguaggio possiamo ad oggi spaziare su tantissimi altri argomenti dalle **Mobile App** ai **Videogiochi**, addirittura con JavaScipt oggi possiamo fare qualcosa che fino a qualche anno fa sembrava impossibile: il _backend_.

Reviewer 3: `Javascript` anzichè `Javascipt`

"Mi stai dicendo che se voglio fare il _Game Developer_ devo prima diventare un **programmatore web**?"
Assolutamente no.

> Se si hanno le idee chiare si da subito allora la scelta del percorso formativo è molto più semplice, se si hanno invece le idee confuse, inizialmente, allora partire dallo sviluppo web è la strada migliore in quanto più accessibile.

Reviewer 3: `sin da subito` anziché `si da subito`

"Ok, ipotizziamo che voglia partire nell'imparare a fare siti web, come scelgo il percorso giusto?"
Non c'è una risposta corretta o sbagliata a questa domanda, tutto dipende dal proprio approccio che si ha con lo studio, ma subentrano anche altri fattori non da poco come ad esempio il tempo (inteso sia come tempo a disposizione per studiare e sia come quantità di tempo necessaria per raggiungere lo scopo), oppure gli sbocchi lavorativi.

## Prima fase: come approcciare lo studio

Ci sono percorsi formativi che lasciano maggiore autonomia e percorsi formativi che invece seguono più assiduamente i propri alunni.
Non c'è un giusto o sbagliato, l'importante è tenere a mente un altro concetto fondamentale: il programmatore non smette mai di imparare, e la maggior parte delle volte impara da solo.
Personalmente mi sento di consigliare, almeno nella fase iniziale, di essere seguiti ed essere curiosi.
Essere seguiti ci permette di avere una guida, un supporto (anche morale e non solo tecnico), qualcuno che indirizza e che spiega all'inizio è molto meglio piuttosto che chiedersi se stiamo approcciando correttamente o meno ad un problema o ad una soluzione.

> Essere curiosi invece ci permette di imparare ad essere autonomi e soprattutto, cosa molto impotante, a trovare la nostra strada.

Reviewer 3: `importante` anziché `impotante`

**Google è il nostro migliore amico**.

## Seconda fase: il tempo

Alcuni percorsi di formazione propongono 3 mesi altri 6 mesi, altri invece hanno dei pacchetti personalizzati.
Alcuni sono costituiti da classi e le lezioni si tengono in presenza, altri invece completamente da remoto con lezioni live oppure on-demand.

> Anche qui non c'è una formula magica.

Reviewer 2: ```suggestion
Anche qui, una formula magica non esiste.
```

> Diventare programmatore non è semplice, essere programmatore è una questione di **mindset**, quindi serve del tempo per cambiare il nostro modo di approcciare e di vedere le cose.

Reviewer 2: Ricordiamoci sempre la gender neutrality, io in primis me la perdo quindi mi permetto di indicarla se la noto ahah

Reviewer 2: Corretto, vero @serenasensini ?

In sostanza... ci vuole tempo.

Quindi maggiore tempo dedichiamo alla nostra formazione e migliore è il risultato che otterremo alla fine del processo.
La scelta dovrebbe essere soggettiva e ponderata, basandosi sulla quantità di tempo che si ha a disposizione.

> Il tempo è il maggiore investimento in questa fase, più tempo dedicate migliore sarà il risultato finale.

Reviewer 2: ```suggestion
Il tempo è il maggiore investimento in questa fase; più tempo dedicate, migliore sarà il risultato finale.
```

"Va bene, quindi quanto tempo serve e quanto tempo dovrei dedicare alla mia formazione?"
Il tempo non è mai abbastanza, quindi la risposta giusta è: il più possibile.

> Ad esempio: se il percorso di formazione offre lezioni mattutine, sarà necessario dedicare almeno altre 3 ore di pratica e approndimenti.

Reviewer 2: ```suggestion
Ad esempio: se il percorso di formazione offre lezioni mattutine, sarà necessario dedicare almeno altre 3 ore di pratica e approfondimenti.
```

> In poche parole: non basta il tempo "vivo" proposto dal percorso di formazione, questo perchè è necessario che chi sta imparando riesca ad "entrare" nell'ottica... ovvero acquisire il **mindset** del programmatore.

Reviewer 2: ```suggestion
In poche parole: non basta il tempo "vivo" proposto dal percorso di formazione, questo perchè è necessario che chi sta imparando riesca ad "entrare" nell'ottica... ovvero acquisire il **mindset** da developer.
```

"Ma se ho un lavoro come faccio ad investire tutto questo tempo?"
Generalmente se ci si affida ad un percorso di formazione per imparare a programmare sarebbe meglio dedicarsi fulltime ad esso, perchè la mole di nozioni da acquisire nel pochissimo tempo a disposizione (3 o 6 mesi) è molto elevata.

> Sarebbe più consigliabile un altro tipo di approccio per chi non dispone di così tanto tempo, ovvero l'autodidatta oppure un percorso persosnalizzato, in questo caso però non sarà possibile ottenere un risultato in soli 3/6 mesi ma ci vorrà molto più tempo.

Reviewer 3: `personalizzato` anziché `persosnalizzato`

I _Web Designer_ di ieri, che si limitavano solamente ad impostare graficamente gli elementi a schermo, sono diventati oggi dei _Frontend Developer_ che sanno mettere mano a _Framework_ e varie _librerie_.

Questo concetto spiega bene 2 cose:

- bisogna continuare ad imparare e ad aggiornarsi
- non si può fare affidamento ad una tecnologia

"Tutto bellissimo, ma come faccio a sapere se un percorso di formazione mi permette di lavorare dopo?
Chi mi da questa certezza dato che spesso e volentieri hanno dei costi alti?"
Il settore della programmazione, come detto in precedenza, non è un settore in perdita e non è un settore saturo.
Basta una piccola ricerca su google per rendersi conto di quanta domanda c'è da parte delle aziende.

Inoltre spesso (anzi oserei dire che è quasi diventato uno "standard") chi propone percorsi di formazione ha a disposizione alcune Aziende Partner che sono interessate ad assumere una volta terminato il percorso.

"Cosa mi posso aspettare al termine di un percorso di formazione? Sarò autonomo e pronto per il mondo del lavoro?"

> Sicuramente a termine di un percorso di formazione si è più consapevoli di cosa faccia un programmatore, di quali siano gli strumenti necessari, di quali siano le qualità che costituiscono un bravo programmatore, e soprattutto di cosa ci si aspetta da lì a poco.

Reviewer 3: `Sicuramente al termine` anziché `Sicuramente a termine`

> # Formazione
> 

Reviewer 2: ```suggestion
```

Rimosso il titolo in quanto non necessario.

Quando si muovono i primi passi, quelli dettati dalla curiosità e dalla voglia di intraprendere un nuovo percorso formativo/lavorativo, ci si imbatte sin da subito con il mondo della **formazione**.

> Basta una semplice ricerca su Google, del tipo: _"Come diventare un developer"_

Reviewer 4: ```suggestion
Basta una semplice ricerca su Google, del tipo: _"Come diventare developer"_
```

> Da lì in avanti i vari canali social, da Instagram a TikTok, dai video promozionali di YouTube alle inserzioni su Facebook, inizieranno a proporre differenti percorsi formativi, principalmente incentrati sulla figura professionale del _Fullstack Developer_.

Reviewer 4: ```suggestion
Da lì in avanti i vari canali social, da Instagram a TikTok, dai video promozionali di YouTube alle inserzioni su Facebook, inizieranno a proporre differenti percorsi formativi, principalmente incentrati sulla figura professionale nota come _Fullstack Developer_.
```

> Ma chi è questo fantomatico _Fullstack Developer_ e come mai è così ricercato dal mercato tanto da essere sulla bocca della maggior parte dei percorsi formativi?

Reviewer 4: ```suggestion
Ma chi è questa fantomatica figura e come mai è così ricercata dal mercato tanto da essere sulla bocca della maggior parte dei percorsi formativi?
```

> Il _Fullstack Developer_ è uno sviluppatore che, teoricamente, conosce uno o più linguaggi di _frontend_ e di \_backend.

Reviewer 4: ```suggestion
Chi ricopre il ruolo di _Fullstack Developer_, teoricamente, conosce uno o più linguaggi di _frontend_ e di \_backend.
```

> Ok, qui si apre un sipario enorme che cercheremmo di ridurre molto semplicemente come segue.

Reviewer 4: ```suggestion
Ok, qui si apre un sipario enorme che cercheremo di ridurre molto semplicemente come segue.
```

Stiamo parlando di **formazione** in ambito **programmazione**, quindi di quali strade si possono percorrere per imparare a programmare, e dunque entrare nel mondo dello sviluppo.
Quindi chi programma crea il programma.
Il programma è composto da una parte grafica ed una parte logica.

Prendiamo l'esempio dell'ambito più semplice (in termini di comprensione): il web.
Un programma sul web è un _sito web_, il quale è composto da una parte grafica, che permette all'utente di interagire, ed una parte logica, che a sua volta interpreta le azioni dell'utente.

Un tempo ormai lontano, fortunatamente, tutto questo veniva svolto da un'unica figura: il _Web Master_.
Poi siamo passati ad avere 2 figure: _Web Designer_ (la figura che si occupava di realizzare l'interfaccia grafica, detta anche UI/UX) ed _Web Developer_ (che traduceva in codice ciò che veniva designato dalla prima figura).

> Oggi la situazione è differente, e per fare un sito web servono almeno 3 figure: _Designer_ (colui che si occupa di realizzare l'interfaccia grafica UI/UX), _Frontend Developer_ e _Backend Developer_.

Reviewer 4: ```suggestion
Oggi la situazione è differente, e per fare un sito web servono almeno 3 figure: _Designer_ (che si occupa di realizzare l'interfaccia grafica UI/UX), _Frontend Developer_ e _Backend Developer_.
```

Tutto ciò ovviamente oggi ha un costo nettamente superiore in termini anche aziendali, per questo motivo il mercato necessita di  _Fullstack Developer_ che siano in grado di tradurre le componenti prodotte dal _Designer_ e agganciare le logiche funzionali del progetto.

Bisogna però che ci addentriamo un pochino meglio nella questione per comprendere bene la differenza tra _Frontend Developer_ e _Backend Developer_.

Chi copre il ruolo di _Frontend Developer_ dovrà programmare tutta la parte "statica" del progetto. Quindi appunto tradurrà in codice tutto ciò che è stato progettato nel _design_.
Invece chi copre il ruolo di _Backend Developer_ dovrà realizzare tutta la parte logica e funzionale, e quindi del comportamento, del progetto.
Possiamo dire che è una spiegazione un pò approssimativa ma per ora dovrebbe bastare.

> Quindi, per sommi capi, il _Fullstack Developer_ è la fusione tra il _Frontend Developer_ ed il _Backend Developer_.

Reviewer 4: ```suggestion
Quindi, per sommi capi, la figuta del _Fullstack Developer_ è la fusione tra _Frontend Developer_ e _Backend Developer_.
```

> I professionisti di settore sono ben consapevoli che ad oggi, con l'evoluzione dei linguaggi di programmazione, risulta **difficilissimo** conoscere e padroneggiare in maniera corretta ed approfondita entrambi gli aspetti (_frontend_ e _backend_), però i percorsi di formazione mirano piuttosto a fornire un'infarinatura generale di entrambi gli aspetti per permettere poi al futuro "developer" di scegliere in autonomia quale strada seguire.

Reviewer 4: ```suggestion
I professionisti di settore sono ben consapevoli che ad oggi, con l'evoluzione dei linguaggi di programmazione, risulta **difficilissimo** conoscere e padroneggiare in maniera corretta ed approfondita entrambi gli aspetti (_frontend_ e _backend_), però i percorsi di formazione mirano piuttosto a fornire un'infarinatura generale di entrambi gli aspetti per permettere poi a chi vuole diventare "developer" di scegliere in autonomia quale strada seguire.
```

La cosa fondamentale che bisogna tenere a mente quando si inizia ad intraprendere un percorso di formazione nell'ambito della programmazione è: se inizi con una cosa non è detto che poi farai quella per tutto il resto della carriera.

"Cosa vuol dire?"
Ad esempio: se inizi come _Frontend Developer_ non è detto che un domani tu non possa cambiare strada e passare al lato oscuro (_backend_).

Per tale motivo spesso chi fa formazione propone dei percorsi di _fullstack_ in ambito web.

"Ok, ma come mai parliamo di siti web e non di videogiochi?"
Perchè la curva di apprendimento è più bassa nel caso dei siti web, quindi è più facile che sia a portata di molta più gente, e se teniamo a mente sempre il concetto fondamentale per cui "se si inizia come _frontend_ non è detto che si finisca da _frontend_" allora vien da sè pensare che un _frontend_ oggi possa diventare un _Game Developer_ domani.

Inoltre le tecnologie che si toccano con mano durante la fase di apprendimento in ambito web sono diverse (HTML, CSS, JavaScript) e ciò permette di avere una overview più ampia.

> Facciamo un esempio: con JavaScript possiamo sviluppare _siti web_ ma non solo, grazie alla versatilità del linguaggio possiamo ad oggi spaziare su tantissimi altri argomenti dalle **Mobile App** ai **Videogiochi**, addirittura con JavaScript oggi possiamo fare qualcosa che fino a qualche anno fa sembrava impossibile: il _backend_.

Reviewer 4: ```suggestion
Facciamo un esempio: con JavaScript possiamo sviluppare _siti web_ ma non solo, grazie alla versatilità del linguaggio possiamo ad oggi spaziare su tantissimi altri argomenti, dalle **Mobile App** ai **Videogiochi**. Addirittura con JavaScript oggi possiamo fare qualcosa che fino a qualche anno fa sembrava impossibile: il _backend_.
```

> "Mi stai dicendo che se voglio fare il _Game Developer_ devo prima diventare un **developer web**?"

Reviewer 4: ```suggestion
"Mi stai dicendo che se voglio diventare _Game Developer_ devo prima formarmi come **developer web**?"
```

La permeanza della tecnologia web permette anche un accesso al mercato più facile e rapido rispetto ad altri campi.

"Mi stai dicendo che è più facile fare i siti web piuttosto che i giochi?"
Assolutamente no.

Il concetto è che fare siti web è più accessibile sia in termini di **risorse tecnologiche** che in termini di **know how** di partenza, per un neofita, rispetto a fare videogiochi.

"Ok, ipotizziamo che voglia partire nell'imparare a fare siti web, come scelgo il percorso giusto?"
Non c'è una risposta corretta o sbagliata a questa domanda, tutto dipende dal proprio approccio che si ha con lo studio, ma subentrano anche altri fattori non da poco come ad esempio il tempo (inteso sia come tempo a disposizione per studiare e sia come quantità di tempo necessaria per raggiungere lo scopo), oppure gli sbocchi lavorativi.

## Prima fase: come approcciare lo studio

Ci sono percorsi formativi che lasciano maggiore autonomia e percorsi formativi che invece seguono più assiduamente i propri alunni.

> Non c'è un giusto o sbagliato, l'importante è tenere a mente un altro concetto fondamentale: il developer non smette mai di imparare, e la maggior parte delle volte impara da solo.

Reviewer 4: ```suggestion
Non c'è un giusto o sbagliato, l'importante è tenere a mente un altro concetto fondamentale: noi developer non smettiamo mai di imparare, e la maggior parte delle volte impariamo per conto nostro.
```

> Sicuramente sarebbe meglio, almeno nella fase iniziale, essere seguiti ed essere curiosi.
> Essere seguiti ci permette di avere una guida, un supporto (anche morale e non solo tecnico), qualcuno che indirizza e che spiega all'inizio può essere d'aiuto piuttosto che chiedersi se stiamo approcciando correttamente o meno ad un problema o ad una soluzione.
> Essere curiosi invece ci permette di imparare ad essere autonomi e soprattutto, cosa molto importante, a trovare la nostra strada.

Reviewer 4: ```suggestion
Sicuramente sarebbe meglio, almeno nella fase iniziale, avere qualcuno che ci segue e sviluppare la curiosità.
Qualcuno che ci segue ci permette di avere una guida, un supporto (anche morale e non solo tecnico), qualcuno che indirizza e che spiega all'inizio può essere d'aiuto piuttosto che chiedersi se stiamo approcciando correttamente o meno ad un problema o ad una soluzione.
La curiosità invece ci insegna l'autonomia e soprattutto, cosa molto importante, ci aiuta a trovare la nostra strada.
```

Facciamo un esempio: settore siti web.
Prima i Layout (la parte grafica del sito web) non seguivano degli "standard" veri e propri, poi ad un certo punto sono nati dei _Framework_ (si tratta di un'insieme di strumenti) che hanno definito tali standard (come ad esempio il concetto di colonne molto utile per rendere un sito web _responsive_ ovvero adattabile ad ogni dispositivo).

Quindi chi progettava i Layout ha dovuto apprendere tali "standard" per non rimanere fuori mercato dato che, ovviamente, tutti i siti ad oggi sono navigabili e fruibili anche dal nostro Smartphone.
I _Web Designer_ di ieri, che si limitavano solamente ad impostare graficamente gli elementi a schermo, sono diventati oggi dei _Frontend Developer_ che sanno mettere mano a _Framework_ e varie _librerie_.

Questo concetto spiega bene 2 cose:

- bisogna continuare ad imparare e ad aggiornarsi
- non si può fare affidamento ad una tecnologia

"Tutto bellissimo, ma come faccio a sapere se un percorso di formazione mi permette di lavorare dopo?

> Chi mi da questa certezza dato che spesso e volentieri hanno dei costi alti?"

Reviewer 4: ```suggestion
Chi mi dà questa certezza dato che spesso e volentieri hanno dei costi alti?"
```

Il settore della programmazione, come detto in precedenza, non è un settore in perdita e non è un settore saturo.

> Basta una piccola ricerca su google per rendersi conto di quanta domanda c'è da parte delle aziende.

Reviewer 4: ```suggestion
Basta una piccola ricerca su Google per rendersi conto di quanta domanda c'è da parte delle aziende.
```

Inoltre spesso (anzi oserei dire che è quasi diventato uno "standard") chi propone percorsi di formazione ha a disposizione alcune Aziende Partner che sono interessate ad assumere una volta terminato il percorso.

> "Cosa mi posso aspettare al termine di un percorso di formazione? Sarò autonomo e pronto per il mondo del lavoro?"

Reviewer 4: ```suggestion
"Cosa mi posso aspettare al termine di un percorso di formazione? Avrò l'autonomia e le competenze necessarie per il mondo del lavoro?"
```

> Sicuramente al termine di un percorso di formazione si è più consapevoli di cosa faccia un developer, di quali siano gli strumenti necessari, di quali siano le qualità che costituiscono un bravo developer, e soprattutto di cosa ci si aspetta da lì a poco.

Reviewer 4: ```suggestion
Sicuramente al termine di un percorso di formazione si è più consapevoli di cosa si potrebbe fare in quanto developer, di quali siano gli strumenti necessari, di quali siano le qualità che costituiscono una figura competente, e soprattutto di cosa ci si aspetta da lì a poco.
```

Sia che il corso duri 3 mesi, o 6, o addirittura 12, si è ancora _Junior_.
Una figura _Junior_ rimane tale fino a quando non ottiene la completa **autonomia** lavorativa, sia con gli strumenti che già conosce e sia con gli strumenti che dovrà apprendere.

> Ricordiamo che l'apprendimento nel mondo della progammazione è una costante, e ricordiamo anche che la maggior parte del tempo, durante la carriera professionale, apprendiamo da soli.

Reviewer 4: ```suggestion
Ricordiamo che l'apprendimento nel mondo della progammazione è una costante, e ricordiamo anche che la maggior parte del tempo, durante la carriera professionale, apprendiamo in solitaria.
```

"Ok tutto chiaro, ma le aziende cosa cercano? Perchè dovrebbero prendere una persona che ha fatto un corso?"
Le aziende cercano, e cercheranno sempre, figure _Junior_ perchè esse sono più "malleabili" e quindi è più facile "impostarle" per quelle che sono le necessità specifiche dell'azienda in questione.

Quando una azienda assume significa che è pronta ad investire del tempo e del denaro su delle nuove risorse, quindi è pronta a fare crescere qualcuno all'interno del proprio ambiente.

Ciò significa 2 cose:

- dovrete ancora imparare
- crescita personale e formativa

Ovviamente però l'azienda ha bisogno di alcune "certezze" prima di assumere un candidato, spesso tali certezze sono difficili da ottenere specialmente riguardo la parte tecnica, non basta un test per capire se si ha davanti la persona adatta per il modus operandi dell'azienda o del team.

Dunque scegliere un percorso di formazione, se si ha la possibilità, sicuramente è meglio piuttosto che decidere di intraprendere un percorso da "autodidatta" sia perchè otterrete dei risultati tangibili in minor tempo, sia anche perchè il percorso di formazione stesso vi farà da "garante".

> Quindi, per concludere, se parti spaesato buttati che strada facendo troverai quella più adatta alle tue esigenze, se invece hai già le idee chiare allora è tutto più semplice (ma ricordati di essere pronto al cambiamento).

Reviewer 4: ```suggestion
Quindi, per concludere, se non sai da dove partire, buttati! Strada facendo troverai quella più adatta alle tue esigenze. Se invece hai già le idee chiare allora è tutto più semplice (ma ricordati di essere disponibile al cambiamento).
```
